### PR TITLE
Add truststore configuration for HTTPS connections to OPA server

### PR DESCRIPTION
**Pull request by Eric84626**
### Type of change
- Enhancement / new feature

### Description
Add truststore configuration for HTTPS connections to OPA server

### Checklist
Related issue: https://github.com/strimzi/strimzi-kafka-operator/issues/8001

### CODE REVIEW

**Comment by Eric84626** [comment]
@scholzj When I built my new code change, I got a checkstyle issue as below.

[ERROR] /home/vsts/work/1/s/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java:639:5: Cyclomatic Complexity is 20 (max allowed is 19). [CyclomaticComplexity]

I think it is very difficult to fix it, would you have some suggestions to fix it?

Thanks

**Comment by scholzj** [comment]
@Eric84626 You can suppress it with an annotation like this:
```
@SuppressWarnings("checkstyle:CyclomaticComplexity")
```
If you search the sources there will be other classes where it is used already.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java [hunk 1]
```diff
@@ -1478,6 +1484,11 @@ private List<VolumeMount> getVolumeMounts() {
             }
         }
 
+        if (authorization instanceof KafkaAuthorizationOpa) {
+            KafkaAuthorizationOpa opaAuthz = (KafkaAuthorizationOpa) authorization;
+            volumeMountList.addAll(AuthenticationUtils.configureOauthCertificateVolumeMounts("authz-opa", opaAuthz.getTlsTrustedCertificates(), OAUTH_TRUSTED_CERTS_BASE_VOLUME_MOUNT + "/authz-opa-certs"));
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java:1489]: Maybe you can rename the `OAUTH_TRUSTED_CERTS_BASE_VOLUME_MOUNT` variable to something more generic as it is not used anymore just for OAuth? E.g. `TRUSTED_CERTS_BASE_VOLUME_MOUNT`?

**Review by scholzj — COMMENTED** [review]
Thanks a lot for this PR. This is a useful addition. I had a quick look and found only one nit. LGTM otherwise. 

But the PR seems to have some commits from other PRs? Can you have a look at it and make sure it has just your changes?

Maybe this is also somethig worth adding to the CHANGELOG.md file?

**Pull request closed without merging**